src/net/interlock.rs

```diff
@@ -0,0 +1,87 @@
+use heapless::String;
+use miniconf::{Error, Miniconf, MiniconfMetadata};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Interlock {
+    /// Set the interlock to armed (true) or disarmed (false).
+    ///
+    /// # Value
+    /// True to arm, false to disarm.
+    pub armed: bool,
+
+    /// Interlock target.
+    /// The Interlock will publish its state (true or false) onto this mqtt path.
+    /// Full path to the desired target. No wildcards.
+    ///
+    /// # Value
+    /// Any string up to 128 characters.
+    pub target: String<128>,
+
+    /// Interlock period in milliseconds.
+    /// The Interlock will publish its state with this period.
+    ///
+    /// # Value
+    /// u64
+    pub period_ms: u64,
+}
+
+impl Miniconf for Interlock {
```

Comment:
Would be good to `impl Miniconf` or `MiniconfAtomic` `for heapless::String`. That would need to live in miniconf though. Could you open an issue?